pylint/checkers/imports.py

```diff
@@ -69,10 +69,12 @@
 from pylint.exceptions import EmptyReportError
 from pylint.graph import DotBackend, get_cycles
 from pylint.interfaces import IAstroidChecker
-from pylint.lint import PyLinter
 from pylint.reporters.ureports.nodes import Paragraph, Section, VerbatimText
 from pylint.utils import IsortDriver, get_global_option
 
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
```

Comment:
Any reason? The direct import seems to have worked previously

Reply:
I'm going to resolve all the issue with TYPE_CHECKING in favor of https://github.com/PyCQA/pylint/pull/5558#discussion_r773948413